### ISSUE TITLE
[FEAT] Support for Cloudinary provider

### ISSUE THREAD

**Issue by SalahAdDin** [muxinc/next-video#223]
We are using Cloudinary on many of our projects for storing media data, it would be nice to have support also for Cloudinary videos in this package.

What do you think?

**Comment by heff** [comment]
Yeah, that'd be a great add. 👍  Would you be interested in helping to build the Provider? You can copy [one of the existing providers](https://github.com/muxinc/next-video/tree/main/src/providers) to get started.

Maybe @tpiros can help from the Cloudinary side?

**Comment by SalahAdDin** [comment]
> Yeah, that'd be a great add. 👍 Would you be interested in helping to build the Provider? You can copy [one of the existing providers](https://github.com/muxinc/next-video/tree/main/src/providers) to get started.
> 
> Maybe @tpiros can help from the Cloudinary side?

Let me check it. 

Thanks

**Comment by SalahAdDin** [comment]
> Yeah, that'd be a great addition. 👍 Would you be interested in helping to build the Provider? You can copy [one of the existing providers](https://github.com/muxinc/next-video/tree/main/src/providers) to get started.
> 
> Maybe @tpiros can help from the Cloudinary side?

Do you have any tutorial about it?

@tpiros, wanna help?

**Comment by SalahAdDin** [comment]
We could base the provider on this [package](https://next.cloudinary.dev/cldvideoplayer/basic-usage).

**Comment by SalahAdDin** [comment]
> Yeah, that'd be a great add. 👍 Would you be interested in helping to build the Provider? You can copy [one of the existing providers](https://github.com/muxinc/next-video/tree/main/src/providers) to get started.
> 
> Maybe @tpiros can help from the Cloudinary side?

@heff , Do you have any tutorial about it?

@tpiros, wanna help?

**Comment by tpiros** [comment]
apologies for the late response. could you elaborate on what you need exactly? thanks!

**Comment by SalahAdDin** [comment]
> apologies for the late response. could you elaborate on what you need exactly? thanks!

adding a cloudinary provider to this package.

**Comment by tpiros** [comment]
@SalahAdDin - thank you for your follow up. At Cloudinary we have our `CldVideoPlayer` component which is compatible for Next.js. You can read more about it [here](https://cloudinary.com/blog/embedding-video-player-next-js-next-cloudinary) and [here](https://next.cloudinary.dev/cldvideoplayer/basic-usage). It's part of our `next-cloudinary` package. Please give it a whirl and let me know if it fits your use cases.

**Comment by SalahAdDin** [comment]
> @SalahAdDin - thank you for your follow up. At Cloudinary we have our `CldVideoPlayer` component which is compatible with Next.js. You can read more about it [here](https://cloudinary.com/blog/embedding-video-player-next-js-next-cloudinary) and [here](https://next.cloudinary.dev/cldvideoplayer/basic-usage). It's part of our `next-cloudinary` package. Please give it a whirl and let me know if it fits your use cases.

I would prefer something more like this package, when using `next-cloudinary` we are tied to use only Cloudinary as a CDN. Plus, that package still uses deprecated API from old NextJS versions.

**Comment by colbyfayock** [comment]
not intending to hijack this thread, but 👋 @SalahAdDin Next Cloudinary maintainer here, what deprecated APIs are you referring to? the package provides support for both the Pages Router and App Router if that's what you're referring to, which the Pages Router is **not** deprecated, otherwise curious what you're referring to

the point of using Next Cloudinary is certainly only with the use case of being a Cloudinary customer, so there's no intent with that package to provide a transferrable library like this if that's what you're looking for, but the benefit then is you can take advantage of more advanced Cloudinary APIs that you can't with this and that aren't available in other providers

**Comment by lh15** [comment]
Does the cloudinary player support server side components?

**Comment by colbyfayock** [comment]
natively, unfortunately not at this time, as i've been struggling to figure out a solution with my current setup to support both cjs and esm. however i have help on the way to address this

for now, you just need to add a very light wrapper around the component

Example of CldImage here: https://next.cloudinary.dev/nextjs-14#app-router-app

to avoid bothering everyone on this thread that might not care about a different Cloudinary package 😂  for anyone looking for help or questions, feel free to create an Issue over here: https://github.com/cloudinary-community/next-cloudinary or DM me somewhere like [Twitter](https://twitter.com/colbyfayock)
